File: app/src/main/java/id/poncoe/latihanmasterandroidponcoe/java/storage/contentprovider/consumerapp/helper/MappingHelper.java
package id.poncoe.latihanmasterandroidponcoe.java.storage.contentprovider.consumerapp.helper;

import android.database.Cursor;

import java.util.ArrayList;

import id.poncoe.latihanmasterandroidponcoe.java.storage.contentprovider.consumerapp.entity.Note;

import static android.provider.BaseColumns._ID;
import static id.poncoe.latihanmasterandroidponcoe.java.storage.contentprovider.consumerapp.db.DatabaseContract.NoteColumns.DATE;
import static id.poncoe.latihanmasterandroidponcoe.java.storage.contentprovider.consumerapp.db.DatabaseContract.NoteColumns.DESCRIPTION;
import static id.poncoe.latihanmasterandroidponcoe.java.storage.contentprovider.consumerapp.db.DatabaseContract.NoteColumns.TITLE;

public class MappingHelper {

    public static ArrayList<Note> mapCursorToArrayList(Cursor notesCursor) {
        ArrayList<Note> notesList = new ArrayList<>();

        while (notesCursor.moveToNext()) {
            int id = notesCursor.getInt(notesCursor.getColumnIndexOrThrow(_ID));
            String title = notesCursor.getString(notesCursor.getColumnIndexOrThrow(TITLE));
            String description = notesCursor.getString(notesCursor.getColumnIndexOrThrow(DESCRIPTION));
            String date = notesCursor.getString(notesCursor.getColumnIndexOrThrow(DATE));
            notesList.add(new Note(id, title, description, date));
        }

        return notesList;
    }

    public static Note mapCursorToObject(Cursor notesCursor) {
        notesCursor.moveToFirst();
        int id = notesCursor.getInt(notesCursor.getColumnIndexOrThrow(_ID));
        String title = notesCursor.getString(notesCursor.getColumnIndexOrThrow(TITLE));
        String description = notesCursor.getString(notesCursor.getColumnIndexOrThrow(DESCRIPTION));
        String date = notesCursor.getString(notesCursor.getColumnIndexOrThrow(DATE));

        return new Note(id, title, description, date);
    }
}

File: app/src/main/java/id/poncoe/latihanmasterandroidponcoe/kotlin/storage/contentprovider/consumerapp/helper/MappingHelper.kt
package id.poncoe.latihanmasterandroidponcoe.kotlin.storage.contentprovider.consumerapp.helper

import android.database.Cursor
import id.poncoe.latihanmasterandroidponcoe.kotlin.storage.contentprovider.consumerapp.db.DatabaseContract.NoteColumns.Companion.DATE
import id.poncoe.latihanmasterandroidponcoe.kotlin.storage.contentprovider.consumerapp.db.DatabaseContract.NoteColumns.Companion.DESCRIPTION
import id.poncoe.latihanmasterandroidponcoe.kotlin.storage.contentprovider.consumerapp.db.DatabaseContract.NoteColumns.Companion.TITLE
import id.poncoe.latihanmasterandroidponcoe.kotlin.storage.contentprovider.consumerapp.db.DatabaseContract.NoteColumns.Companion._ID
import id.poncoe.latihanmasterandroidponcoe.kotlin.storage.contentprovider.consumerapp.entity.Note

import java.util.ArrayList

object MappingHelper {

    fun mapCursorToArrayList(notesCursor: Cursor): ArrayList<Note> {
        val notesList = ArrayList<Note>()

        while (notesCursor.moveToNext()) {
            val id = notesCursor.getInt(notesCursor.getColumnIndexOrThrow(_ID))
            val title = notesCursor.getString(notesCursor.getColumnIndexOrThrow(TITLE))
            val description = notesCursor.getString(notesCursor.getColumnIndexOrThrow(DESCRIPTION))
            val date = notesCursor.getString(notesCursor.getColumnIndexOrThrow(DATE))
            notesList.add(Note(id, title, description, date))
        }

        return notesList
    }

    fun mapCursorToObject(notesCursor: Cursor): Note {
        notesCursor.moveToNext()
        val id = notesCursor.getInt(notesCursor.getColumnIndexOrThrow(_ID))
        val title = notesCursor.getString(notesCursor.getColumnIndexOrThrow(TITLE))
        val description = notesCursor.getString(notesCursor.getColumnIndexOrThrow(DESCRIPTION))
        val date = notesCursor.getString(notesCursor.getColumnIndexOrThrow(DATE))
        return Note(id, title, description, date)
    }
}
